src/tla/modules/raft.tla

```diff
@@ -0,0 +1,394 @@
+\*
+RaftProcessMsg(i, entry) ==
+    IF entry.body.term >= volatileTerm[i]
+    THEN LET raftVarsInit == RaftVars(i)
+             raftVarsTerm ==
+                IF entry.body.term > raftVarsInit.volatileTerm[i]
+                THEN RaftScheduleNewTermVars(raftVars, entry.body.term)
+                ELSE raftVarsInit
+             raftVarsSeen == [raftVarsTerm EXCEPT
+                    !.leaderWitnessMap[entry.replica_id] =
+                        entry.body.is_leader_seen]
+             raftVarsVote ==
+                IF entry.body.vote # 0
+                THEN IF raftVarsSeen.state \in {Follower, Leader}
+                     THEN IF /\ raftVarsSeen.leader = Nil
+                             /\ entry.body.vote # i
+                             /\ entry.body.state = Candidate
+                             /\ raftVarsSeen.volatileVote = Nil
+                             /\ RaftCanVoteFor(i, entry.body.vclock)
+                          THEN RaftScheduleNewVoteVars(raftVarsSeen,
+                                entry.replica_id, entry.body.vclock)
+                          ELSE raftVarsSeen
+                     ELSE \* state = Candidate
+                          IF entry.body.vote = i
+                          THEN LET raftVarsTmp == [raftVarsSeen EXCEPT
+                            !.votesReceived = @ \cup {entry.replica_id}]
+                               IN IF Cardinality(raftVarsTmp.votesReceived) >=
+                                    ElectionQuorum
+                                  THEN RaftBecomeLeaderVars(raftVarsTmp, i)
+                                  ELSE raftVarsTmp
+                          ELSE raftVarsSeen
+                ELSE raftVarsSeen
+             raftVarsFinal == IF entry.body.state # Leader
+                              THEN IF raftVarsVote.leader = entry.replica_id
+                                   THEN [raftVarsVote EXCEPT
+                                            !.leader = Nil,
+                                            !.leaderWitnessMap[i] = FALSE,
+                                            !.isBroadcastScheduled = TRUE]
+                                   ELSE raftVarsVote
+                              ELSE \* entry.body.state = Leader
+                                   IF raftVarsVote.leader # entry.replica_id
+                                   THEN \* raft_sm_follow_leader
+                                        [raftVarsVote EXCEPT
+                                            !.state = Follower,
+                                            !.leader = entry.replica_id,
+                                            !.isBroadcastScheduled = TRUE]
+                                   ELSE raftVarsVote
+         IN RaftApplyVars(i, raftVarsFinal)
+    ELSE UNCHANGED <<raftVars>>
+
+RaftNext(servers) ==
+    \/ \E i \in servers: RaftTimeout(i)
+    \/ \E i \in servers: RaftWorker(i)
+
+================================================================================
```

Comment:
I propose adding a separate specification for Raft and with properties described in Raft paper [^1].

![image](https://github.com/user-attachments/assets/b6554447-8bd2-4835-a749-8df184c89f03)

[^1]: https://raft.github.io/raft.pdf

Reply:
https://github.com/tarantool/tarantool/issues/11374